query/src/main/java/org/infinispan/query/indexmanager/RemoteIndexingBackend.java

```diff
@@ -132,7 +134,13 @@ private void waitForReplacementBackend() {
    }
 
    private void sendCommand(ReplicableCommand command, List<LuceneWork> workList, boolean sync) {
-      rpcManager.invokeRemotely(recipients, command, rpcManager.getDefaultRpcOptions(sync));
+      if (sync) {
+         CompletionStage<Void> completionStage = rpcManager.invokeCommand(recipient, command, VoidResponseCollector.ignoreLeavers(),
+               rpcManager.getSyncRpcOptions());
+         rpcManager.blocking(completionStage);
+      } else {
+         rpcManager.sendTo(recipient, command, DeliverOrder.NONE);
```

Comment:
`DeliverOrder.PER_SENDER`! 
`rpcManager.getDefaultRpcOptions(sync)` with `sync=false` uses `PER_SENDER` instead of `NONE`.